giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/config/AccessTokenRestTemplateConfig.java

```diff
@@ -0,0 +1,54 @@
+package edu.asu.diging.gilesecosystem.web.config;
+
+import java.io.IOException;
+import java.net.URI;
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.client.ClientHttpRequest;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.web.client.RestTemplate;
+
+import com.nimbusds.jose.util.Base64;
+
+import edu.asu.diging.gilesecosystem.util.properties.IPropertiesManager;
+import edu.asu.diging.gilesecosystem.web.service.properties.Properties;
+
+/**
+ * Class to initialize rest template for MITREid connect server
+ * with protected resource access keys
+ * 
+ * @author snilapwa
+ *
+ */
+@Configuration
+public class AccessTokenRestTemplateConfig {
+
+    @Autowired
+    private IPropertiesManager propertyManager;
+
+    @Bean
+    public RestTemplate getAccessTokenRestTemplate() {
```

Comment:
give the bean a name, and then autowire it using the name. this way you could have multiple rest template beans.